lib/src/cmcd/prepareCmcdData.ts

```diff
@@ -41,6 +39,13 @@ export function prepareCmcdData(obj: Record<string, any>, options: CmcdEncodeOpt
 	const reportingMode = options.reportingMode || CMCD_REQUEST_MODE;
 	const keyFilter = version === 1 ? isCmcdV1Key : filterMap[reportingMode];
 
+	if (!options.events || !options.events.includes('rr')){
+		const remainingObj = Object.fromEntries(
+			Object.entries(obj).filter(([key]) => !isCmcdResponseReceivedKey(key)),
+		);
+		obj = remainingObj;
+	}
+
 	// Filter keys based on the version, reporting mode and options
 	let keys = Object.keys(obj).filter(keyFilter);
 
```

Comment:
Two things:

1. I don't think `CmcdEncodeOptions` is the right place for the event list. The `encodeCmcd` function only encodes single records. Event filtering would happen at the report level in the `CmcdReportTarget` object passed to `toCmcdReport`.
2. There is no need to reconstruct the entire object. All of the filtering in `prepareCmcdData` operates on the keys of the data object. Then we iterate over the filtered list of keys one time at the end to construct the final object.

With these in mind, the code would look something like:

```suggestion
	// Filter keys based on the version, reporting mode and options
	let keys = Object.keys(obj).filter(keyFilter);
	
	if (data.e && data.e !== 'rr') {
		keys = keys.filter(key => !isCmcdResponseReceivedKey(key)),
	}
```

That said, we may need to rethink how the filtering works, because there are additional checks we are going to need to perform, like ensuring there is a valid value for `e` if `reportingMode` is `CMCD_EVENT_MODE`. I'm not sure where these new validation steps are going to live, so for now, something like the code above will work.

Reply:
Resolved on https://github.com/qualabs/common-media-library/pull/51/commits/cc0fa05459422536279af491e9a92572a5c9bf05